estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,70 @@
+from dateutil.relativedelta import relativedelta
+from odoo import fields, api, models # type: ignore
+from odoo.exceptions import ValidationError, UserError # type: ignore
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer"
+    _order = "offer_price desc"
+    _sql_constraints = [
+        ('check_offer_price','CHECK(offer_price > 0)','Offer Price must be a positive amount')
+    ]
+
+    #-------------------------------------------Basic Fields-------------------------------------------#
+    offer_price = fields.Float(string="Price", required=True)
+    state = fields.Selection(string="State", selection=[('offer_received','Offer Received'), ('offer_accepted','Offer Accepted'), ('offer_rejected','Offer Rejected')],
+                             copy=False
+                            )
+    validity = fields.Integer(string="Validity(days)", default=7)
+    
+    #-------------------------------------------Relational Fields---------------------------------------#
+    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
+    property_id = fields.Many2one("estate.property", string="Property")
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+    
+    #-------------------------------------------Computed Fields------------------------------------------#
+    deadline_date = fields.Date(string="Deadline", compute="_compute_deadline_date", inverse="_inverse_deadline_date")
+
+    # ------------------------------------------Compute Methods----------------------------------------#
+    @api.depends("create_date", "validity")
+    def _compute_deadline_date(self):
+        for offer in self:
+            date = offer.create_date.date() if offer.create_date else fields.Date.today()
+            offer.deadline_date = date + relativedelta(days=offer.validity)
+
+    def _inverse_deadline_date(self):
+        for offer in self:
+            date = offer.create_date.date() if offer.create_date else fields.Date.today()
+            offer.validity = (offer.deadline_date - date).days
+
+    # --------------------------------------------Action Methods------------------------------------------------
+    def offer_accept_action(self):
```

Comment:
```suggestion
    def action_offer_accept(self):
```
action should be in the starting of the name and change the name accordingly in xml as well